x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -495,12 +485,12 @@ static class GeoTileBucketKeyExtractor implements BucketKeyExtractor {
         public Object value(Object key, String type) {
             assert key instanceof String;
             Rectangle rectangle = GeoTileUtils.toBoundingBox(key.toString());
-            final Map<String, Object> geoShape = new HashMap<>();
+            final Map<String, Object> geoShape = Maps.newLinkedHashMapWithExpectedSize(2);
             geoShape.put(FIELD_TYPE, POLYGON);
             geoShape.put(
                 FIELD_COORDINATES,
                 Collections.singletonList(
-                    Arrays.asList(
+                    List.of(
```

Comment:
Could you also get rid of 2 more `Arrays.asList` usages in this file and replace them with `List.of`?